### ISSUE TITLE
Auto-generated version number causes test_misc.test_deprecate() to fail

### ISSUE THREAD

**Issue by atedstone** [GlacioHack/geoutils#427]
**Describe the bug**
When running `pytest` locally on the latest version of the repo, `TestMisc.test_deprecate` fails with 

```python
ValueError: invalid literal for int() with base 10: 'dev3+g25deaf2'
```

This is caused by the (auto-generated by setuptools?) dev version number.

**To Reproduce**
```bash
git pull upstream main 
git log
---
commit 25deaf26cf18d0e8e34f8fc148ff4d110dcd525c (HEAD -> main, upstream/main)
Author: Romain Hugonnet <snip>
Date:   Fri Jan 5 02:37:48 2024 -0900

     Pin max Python version for PyPI publish (#425)
---
python install -e .
pytest
...
tail -n 2 geoutils/_version.py
__version__ = version = '0.0.18.dev3+g25deaf2.d20240111'
__version_tuple__ = version_tuple = (0, 0, 18, 'dev3', 'g25deaf2.d20240111')
```

**Expected behavior**
All tests should pass.

**System (please complete the following information):**
 - OS: Mac
 - Environment: minimamba with dev-environment.yml

**Additional context**

**Comment by rhugonnet** [comment]
Ah yes, never checked that locally!

More explanations to fix that: 
- `setuptools_scm` is the new tool that manages the version number/file automatically, but it depends on `git` (new version for each commit? or something alike).
- Locally (tracked by `git`), it has therefore a long version number like shown above.
- But when running on CI, `git` is not installed, so the version must be identical to the fixed one used historically, just 0.0.18 for instance.

We're not using the functionality of having "dev" sub-versions at this stage, `setuptools_scm` is simply used to create the version file automatically and ensure a consistent version number across the package (which involved more manual steps before).

**Comment by rhugonnet** [comment]
I'll try to fix it in the afternoon while I still have it in mind
